### DISCUSSION TITLE
Matrix singular!

### DISCUSSION THREAD

**Discussion by Yahiaassam** [OpenModelica/OpenModelica#9341] [Q&A] [answered]
Hello,
I am developing an H2 fill and consume model, I use the boundary component to do the fill and consume. My problem is when I do the padding with a positive input, I get the message "Matrix singular!" at the simulation output, after "The simulation finished successfully." but it gives me problem when I export the FMU.
What caused this message?
How i could fix it?
Thank's for your reply.

**Comment by casella** [comment, marked as answer]
@Yahiaassam this is a message coming from the low-level linear algebra solver. Most likely the simulation involves solving some nonlinear equations with a Newton-Raphson method, and it may happen that the Jacobian becomes singular with a certain choice of initial guess values. Then, the solvers are usually smart and try to cope with that by perturbing the guesses by some epsilons, if you are lucky eventually you get convergence. However, this may cause problems with the FMU (is it CS or ME?).

You can run the simulation with LOG_NLS or LOG_NLS_V and try to figure out what goes wrong. This is currently not very user friendly (understatement), we are working on integrating this feature into the debugger.

**Reply by Yahiaassam** [reply]
@casella it's on co-simu.
haw i can run the  simulation with LOG_NLS or LOG_NLS_V, i never did it before !!

**Reply by casella** [reply]
In OMEdit, Simulation Setup | Simulation Flags you find a wide array of logging options.

**Reply by Yahiaassam** [reply]
Thank you i found it.

**Reply by casella** [reply]
Good luck with that 😃
